### PR TITLE
Provide an Ansible role for VAST

### PR DESCRIPTION
**Pull request by tobim**
This adds a relatively simple role for deploying VAST to a Linux host with Ansible. It supports FHS-based distributions in general, but Debian and derivatives are tested the most.

This role has been in use at Tenzir for a while and we hope to provide additional value to our users by adding it the the open source distribution.

### :memo: Reviewer Checklist

Review this pull request by ensuring the following items:

- [ ] All user-facing changes have changelog entries
- [ ] User-facing changes are reflected on [vast.io](https://github.com/tenzir/vast/tree/master/web)

### CODE REVIEW

File: web/docs/setup/deploy/README.md [hunk 1]
```diff
@@ -6,3 +6,6 @@ it up in a runtime.
 We currently support the following runtimes:
 1. [AWS](deploy/aws)
 2. [Docker](deploy/docker)
+
+We also provide an Ansible role for bare-metal and traditional virtual machines
+in the [ansible directory](/ansible).
```
Comment (dispanser) [inline web/docs/setup/deploy/README.md:11]: We should add a documentation entry point and add `3. [Ansible](deploy/ansible)` to make it more consistent with the other deployment options. Maybe we can embed the existing ansible `README.md`?

**Review by dispanser — APPROVED** [review]
Looks good - I have not actually performed a deployment on my own based on this, assuming that this is mostly open-sourcing what we already have.

If you think that actually running it brings additional value, let me know.

Mid-term, it would be nice if we found a wa to exercise the deployment paths (debian, static) in some form of integration test, but I'm not sure how easy that is (installing into a docker container, maybe?).

File: web/docs/setup/deploy/README.md (hunk 1 above)
Comment (mavam) [reply web/docs/setup/deploy/README.md:11]: I like the idea of `3. [Ansible](deploy/ansible)`. I think this should hold the meat of the documentation, the documentation in the README only pointing to there.

File: ansible/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+This directory contains an Ansible host role for VAST.
+The role can be parameterized with three variables:
```
Comment (dominiklohmann) [inline ansible/README.md:2]: Nit: All throughout the Markdown files, can you reflow text to 80 columns as per our style guide?

Comment (dominiklohmann) [inline ansible/README.md:2]: Please change from passive to active voice.

File: ansible/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+* `config_file`: A path a `vast.yaml` relative to the playbook (required)
+* `read_write_paths`: A list of paths that vast shall be granted access to
```
Comment (dominiklohmann) [inline ansible/README.md:5]:
```suggestion
* `read_write_paths`: A list of paths that VAST shall be granted access to
```

File: ansible/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+This directory contains an Ansible host role for VAST.
+The role can be parameterized with three variables:
+
+* `config_file`: A path a `vast.yaml` relative to the playbook (required)
```
Comment (dominiklohmann) [inline ansible/README.md:4]:
```suggestion
* `config_file`: A path to a `vast.yaml` relative to the playbook (required)
```

File: ansible/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+* `read_write_paths`: A list of paths that vast shall be granted access to
+  in addition to its own state and log directories.
+* `vast_archive`: A tarball of vast structured like those that can be downloaded
```
Comment (dominiklohmann) [inline ansible/README.md:7]:
```suggestion
* `vast_archive`: A tarball of VAST structured like those that can be downloaded
```

File: ansible/README.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  from the [GitHub Releases Page](https://github.com/tenzir/vast/releases).
+  This is used for target distributions that are not based on the `apt`
+  package manager.
+* `vast_debian_package`: A Debian package (`.deb`). This package is used
+  for Debian and Debian based Linux distributions.
+
+#### Example
+
+```
+- name: Deploy vast
```
Comment (dominiklohmann) [inline ansible/README.md:17]:
```suggestion
- name: Deploy VAST
```

File: ansible/roles/vast/tasks/main.yaml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  register: overrides
+- include_tasks: debian.yaml
+  when: ansible_pkg_manager == "apt"
+  register: package
+- include_tasks: static.yaml
+  when: ansible_pkg_manager != "apt"
+  register: package
+- name: Restart if config changed
+  # Handler notification can't express conjuncion or negation, so we use a task.
+  when: config.changed or overrides.changed and not package.changed # noqa no-handler
```
Comment (dominiklohmann) [inline ansible/roles/vast/tasks/main.yaml:42]:
```suggestion
  when: config.changed or overrides.changed and not package.changed  # noqa no-handler
```

To make yamllint happy.

File: ansible/roles/vast/tasks/static.yaml [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+    state: present
+- name: Create vast user
+  ansible.builtin.user:
+    name: vast
+    groups: vast
+    shell: /sbin/nologin
+    append: yes
+    state: present
+    create_home: no
+
```
Comment (dominiklohmann) [inline ansible/roles/vast/tasks/static.yaml:25]: Question/Nit: Is there a general guideline for when you use an empty line between top-level list entries in this YAML? Up to this point there are none, and from here on it's one per step.

File: ansible/roles/vast/tasks/static.yaml [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+    mode: '0755'
+    owner: vast
+    group: vast
+
+- name: Unarchive VAST release tarball
+  ansible.builtin.unarchive:
+    src: "/opt/{{ vast_archive }}"
+    dest: "{{ install_root }}"
+    remote_src: yes
+      # when: vast_download.changed
```
Comment (dominiklohmann) [inline ansible/roles/vast/tasks/static.yaml:71]: Why is this commented out?

File: ansible/roles/vast/tasks/static.yaml [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
+- name: Start vast
```
Comment (dominiklohmann) [inline ansible/roles/vast/tasks/static.yaml:73]:
```suggestion
- name: Start VAST
```

File: changelog/unreleased/changes/2604--ansible-role.md [hunk 11]
```diff
@@ -0,0 +1,2 @@
+VAST now comes with a role definition for Ansible. You can find it directly in
+the `ansible` sub directory.
```
Comment (dominiklohmann) [inline changelog/unreleased/changes/2604--ansible-role.md:2]:
```suggestion
VAST now comes with a role definition for Ansible. You can find it directly in
the `ansible` subdirectory.
```

File: web/docs/setup/deploy/README.md (hunk 1 above)
Comment (dominiklohmann) [inline web/docs/setup/deploy/README.md:11]: This link does not work when you render it locally, as it points to the endpoint `/ansible`, which is not a page we have in our docs.

**Review by dominiklohmann — CHANGES_REQUESTED** [review]

File: web/docs/setup/deploy/README.md (hunk 1 above)
Comment (tobim) [reply web/docs/setup/deploy/README.md]: done.

File: ansible/roles/vast/tasks/static.yaml (hunk 8 above)
Comment (tobim) [reply ansible/roles/vast/tasks/static.yaml:18]: `ansible-lint` doesn't seem to care either way. The samples in official docs separate multi-line YAML array elements with an empty line.

Incidentally `ansible-lint` found a few other issues that I'm going to fix now. Wdyt of adding `ansible-lint` to the style check?

Comment (dominiklohmann) [reply ansible/roles/vast/tasks/static.yaml:18]: Good idea.

Comment (tobim) [reply ansible/roles/vast/tasks/static.yaml:18]: done

**Review by dominiklohmann — APPROVED** [review]

**Pull request merged**